assets/javascripts/comments.js

```diff
@@ -70,102 +70,98 @@ function deleteComment(deleteButton) {
 }
 
 function updateComment(form) {
-  var editorForm = $(form);
-  var url = editorForm.data('put-url');
-  var headingElement = editorForm.find('h4');
-  var markdownElement = editorForm.find('.markdown');
-  var textElement = editorForm.find('[name="text"]');
-  var markdown = markdownElement.html();
-  var text = textElement.val();
-  if (text.length) {
-    textElement.hide();
-    markdownElement.show();
-    markdownElement.html('<em>Loading…</em>');
-    editorForm.find('[name="applyChanges"]').hide();
-    editorForm.find('[name="discardChanges"]').hide();
-    $.ajax({
-      url: url,
-      method: 'PUT',
-      data: editorForm.serialize(),
-      success: function () {
-        // get updated markdown
-        $.ajax({
-          url: url,
-          method: 'GET',
-          success: function (response) {
-            var commentId = headingElement.find('[class="comment-anchor"]')[0].href.split('#comment-')[1];
-            headingElement.replaceWith(renderCommentHeading(response, commentId));
-            textElement.val(response.text);
-            markdownElement.html(response.renderedMarkdown);
-            hideCommentEditor(form);
-          },
-          error: function () {
-            location.reload();
-          }
-        });
-      },
-      error: function (xhr, ajaxOptions, thrownError) {
-        textElement.val(text);
-        markdownElement.html(markdown);
-        showCommentEditor(form);
-        window.alert("The comment couldn't be updated: " + getXhrError(xhr, thrownError));
-      }
-    });
-  } else {
-    window.alert("The comment text mustn't be empty.");
+  const textElement = form.text;
+  const text = textElement.value;
+  if (!text.length) {
+    return window.alert("The comment text mustn't be empty.");
   }
+  const editorForm = $(form);
+  const url = form.dataset.putUrl;
+  const headingElement = editorForm.find('h4');
+  const markdownElement = editorForm.find('.markdown');
+  const markdown = markdownElement.html();
+  textElement.style.display = 'none';
+  markdownElement.show();
+  markdownElement.html('<em>Loading…</em>');
+  editorForm.find('[name="applyChanges"]').hide();
+  editorForm.find('[name="discardChanges"]').hide();
```

Comment:
I suspect we can simplify those, too?

```suggestion
  form.applyChanges.hide();
  form.discardChanges.hide();
```

Reply:
Then it needed to be `.style.display = 'none'` instead of `.hide()` because we're not using jQuery anymore, right? I guess even then your suggestion makes sense, though.

Reply:
I've changed it to use the standard API like most other occurrences to avoid mixing this too much.